server/core/src/access_time_manager.cpp

```diff
@@ -0,0 +1,96 @@
+#include "irods/access_time_manager.hpp"
+
+#include <boost/interprocess/ipc/message_queue.hpp>
+#include <fmt/format.h>
+#include <unistd.h>
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+namespace
+{
+    //
+    // Global Variables
+    //
+
+    // On initialization, holds the PID of the process that initialized the access time manager.
+    // This ensures that only the process that initialized the system can deinitialize it.
+    pid_t g_owner_pid; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+    // Holds the name of the message queue which will exist in shared memory.
+    std::string g_mq_name; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+
+    // The pointer to the message queue which will exist in shared memory.
+    // Allocating on the heap allows us to know when the message queue is constructed/destructed.
+    std::unique_ptr<boost::interprocess::message_queue> g_mq; // NOLINT(cppcoreguidelines-avoid-non-const-global-variables)
+} // anonymous namespace
+
+namespace irods::access_time_manager
+{
+    auto init(const std::string_view _id, std::size_t _queue_size) -> void
+    {
+        if (getpid() == g_owner_pid) {
+            return;
+        }
+
+        using clock_type = std::chrono::system_clock;
+        using seconds = std::chrono::seconds;
+
+        const auto epoch = duration_cast<seconds>(clock_type::now().time_since_epoch()).count();
+        g_mq_name = fmt::format("{}{}_{}", _id, getpid(), epoch);
```

Comment:
you have defaulted the prefix in the database to end with a trailing underscore (`_`).

should we instead just encode the underscore here?

```suggestion
        g_mq_name = fmt::format("{}_{}_{}", _id, getpid(), epoch);
```

and even consider putting epoch first?  so they always sort?

```suggestion
        g_mq_name = fmt::format("{}_{}_{}", _id, epoch, getpid());
```

Reply:
The code originally inserted an underscore in the name, but I removed that to give users full control over the prefix.

Changing the location of the epoch diverges from how the other shared memory filenames are constructed.

Reply:
gotcha - resolving.